arangod/Cluster/AgencyCallbackRegistry.cpp

```diff
@@ -42,25 +44,31 @@ using namespace arangodb;
 
 AgencyCallbackRegistry::AgencyCallbackRegistry(application_features::ApplicationServer& server,
                                                std::string const& callbackBasePath)
-    : _agency(server), _callbackBasePath(callbackBasePath) {}
+  : _agency(server), _callbackBasePath(callbackBasePath) {}
 
 AgencyCallbackRegistry::~AgencyCallbackRegistry() = default;
 
-bool AgencyCallbackRegistry::registerCallback(std::shared_ptr<AgencyCallback> cb) {
-  uint32_t rand;
+bool AgencyCallbackRegistry::registerCallback(std::shared_ptr<AgencyCallback> cb, bool local) {
```

Comment:
As discussed, the object should remember the `local` flag and use it for subsequent operations.